PFBuilder matchAny should take declared input type I, not Object

jypma opened the pull request:
Currently PFBuilder.matchAny takes a lambda with Object as input argument.
This loses type information for partial functions that are not akka
receive functions, e.g. exception handlers (PF<Exception,RouteResult).
With this change, the fallback handler still has the guaranteed type I
visible on its lambda.

ktoso commented:
Refs https://github.com/akka/akka/issues/18996

ktoso commented:
OK TO TEST

ktoso commented:
Looks like a valid fix to me, thanks!
Let's see if validation is happy.

ktoso commented:
Could you also add a test which exemplifies this usage @jypma? Thanks!
Perhaps somewhere in https://github.com/akka/akka/tree/master/akka-actor-tests/src/test/java/akka/japi

jypma committed:
PFBuilder matchAny should take declared input type I, not Object

Currently PFBuilder.matchAny takes a lambda with Object as input argument.
This loses type information for partial functions that are not akka
receive functions, e.g. exception handlers (PF<Exception,RouteResult).
With this change, the fallback handler still has the guaranteed type I
visible on its lambda.

jypma commented:
Test added.

ktoso commented on akka-actor-tests/src/test/java/akka/japi/PFBuilderTest.java:
```diff
@@ -0,0 +1,24 @@
+
+import static org.junit.Assert.*;
+
+public class MatchBuilderTest {
```
`PFBuilderTest`? We don't have `MatchBuilder` AFAIR

jypma replied:
copy/paste oopsie, I'll fix it

ktoso commented:
Hi @jypma, I just noticed this PR is against `release-2.3-dev` which sadly is the wrong branch, as only akka streams and http development is happening on that branch. Can you please re-open a PR against `master` with the same changes? Thanks in advance!

jypma commented:
superseded by https://github.com/akka/akka/pull/18998

ktoso commented:
Thanks
